### PR TITLE
Dockerfiles refactor

### PR DESCRIPTION
**Pull request by francois-travais**
- Use `COPY` instead of `ADD` (see https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy)
- Optimize the `ENV` statements (see https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#env)
- Use consistent UID and GID (here 10000 but it could have bee something else, no problem)
- Reduce the amount of layers and the total size (~50MB for debian image)
- Improve/add `HEALTHCHECK`
- Minor style refactor

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,40 +7,38 @@ ARG TARGET=rel
...
 
 FROM debian:stretch-slim
 
-RUN \
-    apt-get update \
-    && apt-get -y install openssl iproute2 curl jq \
-	&& rm -rf /var/lib/apt/lists/*
+RUN apt-get update && \
+    apt-get -y install openssl iproute2 curl jq && \
+	  rm -rf /var/lib/apt/lists/* && \
```
Comment (codeadict) [inline Dockerfile]: indentation is a bit off here, will be good to keep it all at the same level

File: Dockerfile [hunk 2]
```diff
@@ -7,40 +7,38 @@ ARG TARGET=rel
...
+	  adduser --uid 10000 --system --ingroup vernemq --home /vernemq --disabled-password vernemq
+
+WORKDIR /vernemq
 
 # Defaults
-ENV DOCKER_VERNEMQ_KUBERNETES_LABEL_SELECTOR "app=vernemq"
-ENV DOCKER_VERNEMQ_LOG__CONSOLE console
-ENV PATH "/vernemq/bin:$PATH"
-ADD bin/vernemq.sh /usr/sbin/start_vernemq
+ENV DOCKER_VERNEMQ_KUBERNETES_LABEL_SELECTOR="app=vernemq" \
```
Comment (codeadict) [inline Dockerfile]: perhaps we should merge these with the env on top?

File: Dockerfile.alpine [hunk 3]
```diff
@@ -7,38 +7,36 @@ ARG TARGET=rel
...
+    apk add --no-cache ncurses-libs openssl libstdc++ jq curl bash && \
+    addgroup --gid 10000 vernemq && \
+    adduser --uid 10000 -H -D -G vernemq -h /vernemq vernemq
 
 # Defaults
-ENV DOCKER_VERNEMQ_KUBERNETES_LABEL_SELECTOR "app=vernemq"
-ENV DOCKER_VERNEMQ_LOG__CONSOLE console
-ENV PATH "/vernemq/bin:$PATH"
-ADD bin/vernemq.sh /usr/sbin/start_vernemq
+ENV DOCKER_VERNEMQ_KUBERNETES_LABEL_SELECTOR="app=vernemq" \
```
Comment (codeadict) [inline Dockerfile.alpine]: same as the other Dockerfile

Comment (francois-travais) [reply Dockerfile.alpine]: See above

File: Dockerfile (hunk 2 above)
Comment (codeadict) [reply Dockerfile]: ahh true, sorry missed that in the context of GitHub UI

**Comment by dergraf** [comment]
great stuff. thanks

**Pull request merged**